server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -390,6 +395,9 @@ public Indexer(String indexName,
             this.valueIndexers.add(valueIndexer);
             addGeneratedToVerify(columnConstraints, table, ctxForRefs, ref);
         }
+        for (var ref : table.partitionedByColumns()) {
+            addGeneratedToVerify(columnConstraints, table, ctxForRefs, ref);
+        }
```

Comment:
Another idea, implemented in https://github.com/crate/crate/pull/14305 was to provide all references to the `ColumnIndexWriterProjector` (instead of only non-partitioned ones) but I think we don't need it as we can retrieve it from `partitionNames`.